docs(design): add autoware design overview (#86)
autowarefoundation/autoware-documentation: docs/design/index.md

    Under Construction

> ### Autoware architecture

Reviewer: Mm, this should be at least after the `Autoware concepts` section.
And it might be placed under the `autoware-interfaces/` directory.

I'd like to ask other people about where to put these documents.

@xmfcx @mitsudome-r @isamu-takagi @LalithVipulananthan Where do you think is the best place?

Reviewer 2: @kenji-miyake @TakaHoribe 
> And it might be placed under the autoware-interfaces/ directory.

I think `design/autoware-architecture/<component-name>/` is an overview and `design/autoware-interfaces/components/<component-name>/` is a detail. So the architecture focuses on what components and modules are, how they are connected, what the purpose is, and so on. On the other hand, Interface describes input and output for each components or module.

For example, in the architecture section, there are perception and planning components. And they are sending and receiving data of predicted objects to avoid or stop. Here, a predicted object is just a predicted object, it`s data structure is not explained unless it is necessary. The interface section describes that predicted object has header, uuid, classification, kinematics, etc. like [this page](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-interfaces/components/planning/#inputs).

Also, the component interface is for the public interface. The internal messages are described elsewhere, such as in the README.

Reviewer: @isamu-takagi Thank you for your comment.
So is this good?

Autoware architecture
- target readers: beginner developers/users
- purpose: give them a high-level understanding

Autoware interfaces
- target readers: accustomed developers
- purpose: give them the information of how-to-develop

Reviewer 3: I think this is a good way to organise the content.  👍🏾

Author: I updated the document. How's this?

@LalithVipulananthan I would appreciate it if you could make modifications as needed.

Reviewer 3: @TakaHoribe Looks fine to me!

### Autoware concepts

> The [Autoware concepts page](autoware-concepts/index.md) describes the "design philosophy" of Autoware. Readers (service providers and all Autoware users) will know the basic concepts underlying Autoware development, such as microautonomy and core/universe architecture.

Reviewer 3: I don't think there is any need for the double quotes around `design philosophy`, `overview` nor `detailed`.

```suggestion
The [Autoware concepts page](autoware-concepts/index.md) describes the design philosophy of Autoware. Readers (service providers and all Autoware users) will learn the basic concepts underlying Autoware development, such as microautonomy and the Core/Universe architecture.
```

Author: Thank you. Addressed.

### Autoware architecture

> The [Autoware architecture page](autoware-architecture/index.md) describes an "overview" of each module that composes the Autoware. Readers (all Autoware users) will know a high-level picture of how each module that composes Autoware works.

Reviewer 3: ```suggestion
The [Autoware architecture page](autoware-architecture/index.md) describes an overview of each module that makes up Autoware. Readers (all Autoware users) will gain a high-level picture of how each module that composes Autoware works.
```

Author: Addressed.

### Autoware interfaces

> The [Autoware interfaces page](autoware-interfaces/index.md) describes a "detailed" interface of each module that composes the Autoware. Readers (intermediate developers) will know how to add a new functionality to Autoware and connect their own modules with the Autoware.

Reviewer 3: ```suggestion
The [Autoware interfaces page](autoware-interfaces/index.md) describes in detail the interface of each module that makes up Autoware. Readers (intermediate developers) will learn how to add new functionality to Autoware and how to integrate their own modules with Autoware.
```

Author: Addressed.

### Autoware concepts

The concept of the Autoware is available [here](autoware-concepts/index.md).

> - Target readers:
> - Purpose:

Reviewer: If it's empty, to avoid confusion:

```suggestion
TODO(TIER IV): Add target readers and purpose.
```

Reviewer: Of course, it's better to write something.

@mitsudome-r @LalithVipulananthan Any ideas? Other sections are written as the attached image.

![image](https://user-images.githubusercontent.com/31987104/167606074-97a49869-d726-4dea-991a-80a22a824079.png)

Reviewer: I think we can delete it for now.
https://github.com/autowarefoundation/autoware-documentation/pull/86/files#r869934472

Reviewer: ```suggestion
```

Author: Addressed.

### Autoware architecture

The overall architecture of Autoware and design overview of each component are available [here](autoware-architecture/index.md).

> - Target readers: all users and beginner developers
> - Purpose: describe high-level understanding
> 
> ### Autoware interfaces
> 
> The detailed interface design of each component are available [here](autoware-interfaces/index.md).
> 
> - Target readers: intermediate developers
> - Purpose: describe the information of how-to-develop

Reviewer: By the way, since this is just my comment for confirmation, I think you can delete it.
https://github.com/autowarefoundation/autoware-documentation/pull/86/files#r856877533

Instead, we can write several sentences in other PRs.

```suggestion
### Autoware interfaces

The detailed interface design of each component are available [here](autoware-interfaces/index.md).
```
